andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockitoWhen.java

```diff
@@ -32,30 +33,59 @@ public MockitoWhen(String methodToSetUp, Comparison comparison, int expectedNumb
     public boolean visit(MethodInvocation mi) {
 
         String methodName = mi.getName().toString();
-
         /**
          * As soon as a Mockito's when happen, we wait for the next method call.
          * We know that a when() method was just called because of the inWhenMode variable flag.
          *
          * If the method call just after when() matches the one we are expecting, bingo!
          */
         if(inWhenMode) {
-            if(methodToSetUp.equals(methodName)) {
+            if(methodToSetUp.equals(methodName) || methodToSetUp.equals(previousMethodName)) {
+                // With the normal syntax, if we encounter a match in method name,
+                // it will be right after when(...).
+                // Example: when(mockedList.add("2")).thenReturn(true);
+                // the order in which they will be processed is thenReturn(...) -> when(...) -> add(...)
+                // With the other syntax we can encounter a match either before when or after it
+                // Example: doNothing().when(mockedList).someVoidMethod()
+                // the order will be someVoidMethod() -> when(...) -> doNothing()
+                // Alternatively, we can also write doReturn(true).when(mockedList.add("1"))
+                // the order will be when(...) -> doReturn(...) -> add(...)
+                // So we check if we encounter match on the current command or on the previous
                 numberOfCallsToWhen++;
+                inWhenMode = false;
+                previousMethodName = "";
+            } else if(previousMethodName.contains("do") && !methodName.equals("when")){
```

Comment:
Do we need to be specific here, instead of `contains("do")`, more like "equals("doNothing")` or is that not a problem?

Reply:
I see it's because `doReturn` and `doNothing`. I think it's best to be super precise here. If you extract to a private method, it becomes easier to compare for both.

Reply:
I totally agree, I added a list, which contains some doWhen invocations. If there is any need to add more, it is very simple to do so, by adding them to the list.